mos_tests/neutron/python_tests/test_l3_ha.py

```diff
@@ -435,3 +434,49 @@ def test_ban_l3_agent_for_many_routers(self, variables):
                 node_to_ban = new_agent['host']
 
         assert ping_result['sended'] - ping_result['received'] < 10
+
+    def test_ban_active_l3_agent_with_external_connectivity(self, router,
+                                                            prepare_openstack):
+        """Ban l3-agent with ACTIVE ha_state for router and check external ping
+
+         Steps:
+            1. Create network net01, subnet net01_subnet
+            2. Create router with gateway to external net and
+               interface with net01
+            3. Launch instance and associate floating IP
+            4. Check ping from instance to google DNS
+            5. Ban active l3 agent
+            6. Wait until router rescheduled
+            7. Stop ping
+            8. Check that ping lost less than 40 packets
+        """
+        # collect l3 agents and group it by ha_state
+        agents = defaultdict(list)
```

Comment:
I beleive you don't need to check count of active and standby agents in this case.